src/app/cluster/details/external-cluster/update-external-cluster-machine-deployment-dialog/component.ts

```diff
@@ -108,7 +120,12 @@ export class UpdateExternalClusterMachineDeploymentDialogComponent extends BaseF
     this._externalMachineDeploymentService
       .machineDeploymentUpgrades(this.data.projectID, this.data.clusterID, this.data.machineDeployment?.id)
       .pipe(take(1))
-      .subscribe(this._setDefaultVersion.bind(this));
+      .subscribe((upgrades: MasterVersion[]) => {
+        this._setDefaultVersion(upgrades);
+        this.versionsLoaded = !!upgrades?.length;
```

Comment:
IMO there's no need for `versionsLoaded` property and we can use `upgrades?.length` directly.